transactions/internal/service/transaction_manager.go

```diff
@@ -0,0 +1,88 @@
+package service
+
+import (
+	"context"
+	"errors"
+
+	grpc_gen "github.com/GenesisEducationKyiv/software-engineering-school-4-0-boredgus/transactions/internal/clients/gen"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-boredgus/transactions/internal/config"
+)
+
+type (
+	CustomerService interface {
+		CreateCustomer(ctx context.Context, email string) error
+		CreateCustomerRevert(ctx context.Context, email string) error
+	}
+	DispatchService interface {
+		SubscribeForDispatch(ctx context.Context, email, dispatchID string) (*grpc_gen.Subscription, error)
+		UnsubscribeFromDispatch(ctx context.Context, email, dispatchID string) (*grpc_gen.Subscription, error)
+	}
+	Broker interface {
+		Publish(msg interface{})
+	}
+	transactionManager struct {
+		transactionManagerURL string
+		customerService       CustomerService
+		dispatchService       DispatchService
+		broker                Broker
+		logger                config.Logger
+	}
+)
+
+func NewTransactionManager(
+	transactionManagerURL string,
+	customerService CustomerService,
+	dispatchService DispatchService,
+	broker Broker,
+	logger config.Logger,
+) *transactionManager {
+	return &transactionManager{
+		transactionManagerURL: transactionManagerURL,
+		customerService:       customerService,
+		dispatchService:       dispatchService,
+		broker:                broker,
+		logger:                logger,
+	}
+}
+
+var (
+	ErrAlreadyExists = errors.New("already exists")
+	ErrNotFound      = errors.New("not found")
+)
+
+func (m *transactionManager) SubscribeForDispatch(ctx context.Context, email, dispatchID string) error {
+	err := m.customerService.CreateCustomer(ctx, email)
+	if err != nil && !errors.Is(err, ErrAlreadyExists) {
+		m.logger.Errorf("failed to create a customer: %v", err)
+
+		return err
+	}
+
+	subscription, err := m.dispatchService.SubscribeForDispatch(ctx, email, dispatchID)
+	if err != nil {
+		m.logger.Errorf("failed to subscribe for a dispatch: %v", err)
+
+		if err := m.customerService.CreateCustomerRevert(ctx, email); err != nil {
+			m.logger.Errorf("failed to revert creation of customer: %v", err)
+		}
+
+		return err
+	}
+
+	m.broker.Publish(subscription)
+
+	return nil
+}
```

Comment:
Left comments in private messages in telegram. Resolve this after we will be on the same point

Reply:
Questions that have been discussed: 
1. Why had I chosen synchronous way to implement SAGA
2. Why is there no retry logic

Result of our conversation:
1. We decided to left synchronous implementation of SAGA.
2. I added retry logic when transport error occurs.

Reply:
Good job 
And there left some points for discussion =)